lib/kafka/async_producer.rb

```diff
@@ -218,6 +218,12 @@ def run
             raise "Unknown operation #{operation.inspect}"
           end
         end
+      rescue Kafka::Error => e
+        @logger.error "Unexpected Kafka error #{e.class}: #{e.message}\n#{e.backtrace.join("\n")}"
+        @logger.info "Restarting in 10 seconds..."
+
+        sleep 10
+        retry
```

Comment:
Should we have a retry limit here in case the error is related to a problem with the message and not the connection? Example: https://github.com/abrandoned/protobuf-nats/blob/master/lib/protobuf/nats/client.rb#L96-L105

Reply:
Also, 10 seconds might be a bit much. You could make this 1 second and it would still use a really low amount of system resources.

Reply:
I think it's fairly typical to have a high number of e.g. Unicorn worker processes running on a host, and this would run per-worker – so if there's a systemic error, you'd get a restart per worker based on the frequency. I think 10s should suffice – this would only cover totally unforeseen errors, as we should add specific handling of errors whenever possible.

Reply:
Interesting. We run on jruby with puma backed by a single thread pool, so for us one instance has one async producer. Would you be open to making this configurable? I'm happy to make a follow-up change :)